qiskit/optimization/algorithms/minimum_eigen_optimizer.py

```diff
@@ -30,16 +31,18 @@ class MinimumEigenOptimizationResult(OptimizationResult):
     def __init__(self, x: Union[List[float], np.ndarray], fval: float,
                  variables: List[Variable],
                  samples: List[Tuple[str, float, float]],
-                 min_eigen_solver_result: Optional[MinimumEigensolverResult] = None) -> None:
+                 min_eigen_solver_result: Optional[MinimumEigensolverResult] = None,
+                 status: OptimizationResultStatus = OptimizationResultStatus.SUCCESS) -> None:
```

Comment:
Please make `status` parameter required.

Reply:
I am not sure if something like `Required` is available in typing provided by python. https://github.com/python/typing/blob/5d2f3b1992b8b66d41fe001df148693b45d30e31/src/typing.py#L23-#L98

Reply:
I do not think there is something like `Required` inside the typing provided by python
https://github.com/python/typing/blob/5d2f3b1992b8b66d41fe001df148693b45d30e31/src/typing.py#L23-L98

Reply:
https://github.com/python/typing/blob/5d2f3b1992b8b66d41fe001df148693b45d30e31/src/typing.py#L23-L30

Reply:
I meant, delete default assignment to `status`, this causes the parameter required, you have to pass a value for this parameter.

Reply:
Correct, but I do need to define a default value for it. Should I go with `None`?

Reply:
No, no default value, that's the thing.

Reply:
Then it throws an error: `default value must be specified here Python(parser-16)`

Reply:
Non default(required) parameters go first. So, move it up on the list of parameters, we still can do this.